src/components/ui/AlertDialog/fragments/AlertDialogTitle.tsx

```
@@ -10,11 +11,46 @@ type PrimitiveH2Props = React.ComponentPropsWithoutRef<typeof Primitive.h2>;
-const AlertDialogTitle = forwardRef<AlertDialogTitleElement, AlertDialogTitleProps>(({ children, className = '', ...props }, ref) => {
-    const { rootClass } = useContext(AlertDialogContext);
-    return <Primitive.h2 ref={ref} className={`${rootClass}-title ${className}`} {...props}>{children}</Primitive.h2>;
+const AlertDialogTitle = forwardRef<AlertDialogTitleElement, AlertDialogTitleProps>(({
+    children,
+    className = '',
+    asChild = false,
+    id,
+    ...props
+}, ref) => {
+    const { rootClass, setTitleId, titleId: currentTitleId } = useContext(AlertDialogContext);
+    const titleId = id ?? Floater.useId();
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)